blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -64,14 +65,18 @@ public JSONMetricsContainer(String tenantId, List<JSONMetric> metrics) {
         delayedMetrics = new ArrayList<Metric>();
         for (JSONMetric jsonMetric : jsonMetrics) {
             Locator locator;
+            boolean isValid;
             if (jsonMetric instanceof ScopedJSONMetric) {
                 ScopedJSONMetric scopedMetric = (ScopedJSONMetric)jsonMetric;
                 locator = Locator.createLocatorFromPathComponents(scopedMetric.getTenantId(), jsonMetric.getMetricName());
+                isValid = scopedMetric.isValid();
             } else {
                 locator = Locator.createLocatorFromPathComponents(tenantId, jsonMetric.getMetricName());
+                isValid = jsonMetric.isValid();
             }
 
-            if (jsonMetric.getMetricValue() != null) {
+            // convert and add metric only if it's valid (no validation error)
+            if ((jsonMetric.getMetricValue() != null) && isValid ) {
```

Comment:
``toMetrics()`` method now has a side-affect that it will only return valid metrics. May be we can create a separate method to get only valid metrics.